service/controller/resource/clusterconfigmap/desired.go

```diff
@@ -67,56 +68,55 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*cor
 		clusterCA = string(secret.Data["tls.crt"])
 	}
 
-	appOperatorValues := map[string]interface{}{
-		"app": map[string]interface{}{
-			"watchNamespace":    cr.GetNamespace(),
-			"workloadClusterID": key.ClusterID(&cr),
-		},
-		"provider": map[string]interface{}{
-			"kind": r.provider,
-		},
-		"registry": map[string]interface{}{
-			"domain": r.registryDomain,
-		},
-	}
+	// By default, the IP for the coredns Service belongs to the pod CIDR,
+	// but if "serviceSubnet" is set in KubeadmControlPlane CR then we want to override it with an IP belonging
+	// to the service CIDR.
+	var clusterDNSIP = r.dnsIP
+	{
+		if cr.Spec.ControlPlaneRef != nil {
+			kubeadmControlPlane := &unstructured.Unstructured{}
+			kubeadmControlPlane.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   cr.Spec.ControlPlaneRef.GroupVersionKind().Group,
+				Kind:    cr.Spec.ControlPlaneRef.Kind,
+				Version: cr.Spec.ControlPlaneRef.GroupVersionKind().Version,
+			})
+			err = r.k8sClient.CtrlClient().Get(ctx, client.ObjectKey{
+				Namespace: cr.Namespace,
+				Name:      cr.Spec.ControlPlaneRef.Name,
+			}, kubeadmControlPlane)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
 
-	clusterValues := map[string]interface{}{
-		"baseDomain": key.BaseDomain(&cr, r.baseDomain),
-		"chartOperator": map[string]interface{}{
-			"cni": map[string]interface{}{
-				"install": true,
-			},
-		},
-		"cluster": map[string]interface{}{
-			"calico": map[string]interface{}{
-				"CIDR": podCIDR,
-			},
-			"kubernetes": map[string]interface{}{
-				"API": map[string]interface{}{
-					"clusterIPRange": r.clusterIPRange,
-				},
-				"DNS": map[string]interface{}{
-					"IP": r.dnsIP,
-				},
-			},
-		},
-		"clusterCA":    clusterCA,
-		"clusterDNSIP": r.dnsIP,
-		"clusterID":    key.ClusterID(&cr),
-		"clusterCIDR":  "",
-		"provider":     "unknown",
+			serviceCidr, serviceCidrFound, err := unstructured.NestedString(kubeadmControlPlane.Object, []string{"spec", "kubeadmConfigSpec", "clusterConfiguration", "networking", "serviceSubnet"}...)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+
+			if !serviceCidrFound {
+				serviceCidr = DefaultK8sServiceCidr
```

Comment:
I am not sure we should define defaults in the app, that sounds dangerous, I would rather error here

Reply:
or at least this value for the default CIDR should be taken from the config repo and passed to the operator as a value

Reply:
The constant `DefaultK8sServiceCidr` contains the default service CIDR used by kubernetes. That means it's the CIDR that will be used when that field is not set. Erroring here when the field is not set means that the operator would fail for all the clusters that don't set this CIDR on their CR, which is most of them at the moment.

Currently the operator receives the service CIDR [through a parameter called `clusterIPRange`](https://github.com/giantswarm/cluster-apps-operator/blob/master/helm/cluster-apps-operator/values.yaml#L32). We could change that to be the kubernetes default, but that would mean we are changing the default service CIDR used for apps like aws-operator, azure-operator, etc. I can't tell the potential implications of such a change. Would apps be re-installed to use the new value? would clusters be rolled? (`k8scloudconfig` relies on this value). Do you know @giantswarm/team-phoenix ?

Reply:
cluster-apps-operator is only used by CAPI cluster or I am wrong?
we always set the service cidr range in the vintage product  to value that is specified In the config repo for example https://github.com/giantswarm/config/blob/main/installations/anubis/config.yaml.patch#L57

but in vintage its handled by provider operator and k8scloudconfig not by  cluster-operator or cluster-apps-operator

Reply:
yeah, you are right, we only use this on CAPI. Ok, I'll change the default at the config level then if you prefer.